code/game/objects/items/RSF.dm

```diff
@@ -8,178 +8,201 @@ RSF
-	var/matter = 10
-	var/toxin = 0
+	spent_icon_state = "rcd"
+	var/toxin = FALSE
 	var/cooldown = 0
 	var/cooldowndelay = 10
-	w_class = WEIGHT_CLASS_NORMAL
+	max_matter = 10
+	cost_by_item = list()
+	dispense_cost = 100
+	discriptor = "cookie-units"
+	action_type = "Fabricates"
 
-/obj/item/cookiesynth/examine(mob/user)
+/obj/item/rsf/cookiesynth/Initialize()
 	. = ..()
-	. += "<span class='notice'>It currently holds [matter]/10 cookie-units.</span>"
+	START_PROCESSING(SSprocessing, src)
 
-/obj/item/cookiesynth/attackby()
+/obj/item/rsf/cookiesynth/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/obj/item/rsf/cookiesynth/attackby()
 	return
 
-/obj/item/cookiesynth/emag_act(mob/user)
+/obj/item/rsf/cookiesynth/emag_act(mob/user)
 	obj_flags ^= EMAGGED
 	if(obj_flags & EMAGGED)
 		to_chat(user, "<span class='warning'>You short out [src]'s reagent safety checker!</span>")
 	else
 		to_chat(user, "<span class='warning'>You reset [src]'s reagent safety checker!</span>")
-		toxin = 0
 
-/obj/item/cookiesynth/attack_self(mob/user)
+/obj/item/rsf/cookiesynth/attack_self(mob/user)
 	var/mob/living/silicon/robot/P = null
 	if(iscyborg(user))
 		P = user
-	if((obj_flags & EMAGGED)&&!toxin)
-		toxin = 1
-		to_chat(user, "<span class='alert'>Cookie Synthesizer hacked.</span>")
-	else if(P.emagged&&!toxin)
-		toxin = 1
+	if(((obj_flags & EMAGGED) || (P && P.emagged)) && !toxin)
+		toxin = TRUE
+		to_dispense = /obj/item/reagent_containers/food/snacks/cookie/sleepy
 		to_chat(user, "<span class='alert'>Cookie Synthesizer hacked.</span>")
 	else
-		toxin = 0
+		toxin = FALSE
+		to_dispense = /obj/item/reagent_containers/food/snacks/cookie
 		to_chat(user, "<span class='notice'>Cookie Synthesizer reset.</span>")
 
-/obj/item/cookiesynth/process()
-	if(matter < 10)
-		matter++
+/obj/item/rsf/cookiesynth/process()
+	matter = min(matter + 1, max_matter) //We add 1 up to a point
 
-/obj/item/cookiesynth/afterattack(atom/A, mob/user, proximity)
-	. = ..()
+/obj/item/rsf/cookiesynth/afterattack(atom/A, mob/user, proximity)
 	if(cooldown > world.time)
 		return
-	if(!proximity)
-		return
-	if (!(istype(A, /obj/structure/table) || isfloorturf(A)))
-		return
-	if(matter < 1)
-		to_chat(user, "<span class='warning'>[src] doesn't have enough matter left. Wait for it to recharge!</span>")
-		return
-	if(iscyborg(user))
-		var/mob/living/silicon/robot/R = user
-		if(!R.cell || R.cell.charge < 400)
-			to_chat(user, "<span class='warning'>You do not have enough power to use [src].</span>")
-			return
-	var/turf/T = get_turf(A)
-	playsound(src.loc, 'sound/machines/click.ogg', 10, TRUE)
-	to_chat(user, "<span class='notice'>Fabricating Cookie...</span>")
-	var/obj/item/reagent_containers/food/snacks/cookie/S = new /obj/item/reagent_containers/food/snacks/cookie(T)
-	if(toxin)
-		S.reagents.add_reagent(/datum/reagent/toxin/chloralhydrate, 10)
-	if (iscyborg(user))
-		var/mob/living/silicon/robot/R = user
-		R.cell.charge -= 100
-	else
-		matter--
+	. = ..()
```

Comment:
So, what is this cooldown used for now?

Reply:
it's a limiter on the cookie output of the thing. It's actually a bit too long, as it outstrips the process, and makes the whole recharging bit moot.